### PR TITLE
Update pna.p4

### PR DESCRIPTION
**Pull request by mariobaldi**
Changes to reflect architectural changes of PR #100

### CODE REVIEW

**Review by thomascalvert-xlnx — APPROVED** [review]
LGTM with the minor capitalisation fix. In the interest of future archeologists, perhaps the commit message could be a bit more descriptive (e.g. _pna.p4: changes for flex routing_) ? Otherwise a great companion to #100.

File: pna.p4 [hunk 1]
```diff
@@ -556,46 +556,34 @@ extern Digest<T> {
...
     // the id of the network port on which the packet arrived.
-    // If this packet has direction HOST_TO_NET, input_port contains
+    // If this packet has FROM_HOST source, input_port contains
     // the id of the vport from which the packet came
-    PortId_t                 input_port;   // network port id
+    PortId_t                 input_port;   // network/host port id
 }
 
+extern bool Is_host_port (PortID_t p);
+extern bool Is_net_port (PortID_t p);
```
[the remark this answers is not in the document]
Comment (loalan) [reply pna.p4:578]: Also, all the extern functions are lower case (so "is_host_port" instead of "Is_host_port")

File: pna.p4 [hunk 2]
```diff
@@ -556,46 +556,34 @@ extern Digest<T> {
...
     // common fields initialized for all packets that are input to main
     // parser, regardless of direction.
-    PNA_Direction_t          direction;
-    PassNumber_t             pass;
-    bool                     loopedback;
+    bool                     recirculated;
     Timestamp_t              timestamp;
     ParserError_t            parser_error;
-    ClassOfService_t         class_of_service;
+    ClassOfService_t         class_of_service; //TODO: is this to stay?
```
Comment (loalan) [inline pna.p4:586]: Suggest we leave out TODOs comments

File: pna.p4 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jafingerhut) [reply pna.p4:578]: To try to get through the release process more quickly, I will approve and merge this as is, and address these comments in a separate PR.

**Review by jafingerhut — APPROVED** [review]
LGTM, will fix some minor things in a future PR later today

File: pna.p4 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jafingerhut) [reply pna.p4:578]: Both of these changes, and many more, in PR https://github.com/p4lang/pna/pull/105

File: pna.p4 (hunk 2 above)
Comment (jafingerhut) [reply pna.p4:586]: Changed in https://github.com/p4lang/pna/pull/105

**Pull request merged**